app/controllers/invoices_controller.rb

```diff
@@ -1,20 +1,24 @@
 class InvoicesController < ApplicationController
-    before_action :set_invoice, only: :show
-  
-    def index
-      @invoices = Invoice.all
-      render json: @invoices
-    end
-  
-    def show
-      render json: @invoice
-    end
-  
-    private
-  
-    def set_invoice
-      @invoice = Invoice.find(params[:id])
-    rescue ActiveRecord::RecordNotFound
-      render json: { error: "Invoice not found" }, status: :not_found
-    end
+  before_action :set_invoice, only: [:show]
+
+  rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+
+  def index
+    @invoices = Invoice.all
+    render json: @invoices.map { |invoice| InvoiceSerializer.new(invoice).serialize }
+  end
+
+  def show
+    render json: InvoiceSerializer.new(@invoice).serialize
+  end
+
+  private
+
+  def set_invoice
+    @invoice = Invoice.find(params[:id])
+  end
+
+  def record_not_found
+    render json: { error: 'Invoice not found' }, status: :not_found
```

Comment:
No tratamento da exception que chama esse método, vc pode usar o I18N para padronizar o texto gerado quando o método for chamado independente do model que esteja dando erro.